foundations/javascript_basics/fundamentals-5.md

```diff
@@ -6,30 +6,157 @@ Congratulations on making it to one of the last lessons in Foundations! By this
+Consider the array from our previous example, `[1, 2, 3, 4, 5]`. 
+If we wanted to remove all even numbers from this array, we could use `.filter()` like this:
+
+```js
+arr.filter(isOdd); // Outputs [1, 3, 5];
+```
+
+- `filter` will iterate through `arr` and pass **every value** into the `isOdd` callback function, one at a time. 
+- `isOdd` will return `true` when the value is odd, which means this value is included in the output.
+- If it's an even number, `isOdd` will return `false` and not include it in the final output.
+
+#### The reduce method
+
+Finally, let's say that we wanted to multiply all of the numbers in our `arr` together like this: `1 * 2 * 3 * 4 * 5`.
+First, we'd have to declare a variable `total` and initialize it to 1. Then, we'd iterate through the array with a `for` loop and multiply the `total` by the current number.
+
+But we don't actually need to do all of that, we have our `reduce` method that will do the job for us. Just like `.map()` and `.reduce()` it expects a callback function.
+However, there are two key differences with this array method:
+- The callback function takes two arguments instead of one. The first argument is the `accumulator`, which is the current value of the result *at that point in the loop*. The first time through, this value will either be set to the `initialValue` (described in the next bullet point), or the first element in the array if no `initialValue` is provided. The second argument for the callback is the `current` value, which is the item currently being iterated on.
+- It also takes in an `initialValue` as a second argument (after the callback), which helps when we don't want our initial value to be the first element in the array. For instance, if we wanted to sum all numbers in an array, we could call reduce without an `initialValue`, but if we wanted to sum all numbers in an array and add 10, we could use 10 as our `initialValue`.
+
+```js
+arr.reduce(((total, currentItem) => total * currentItem), 1); // Outputs [1, 2, 6, 24, 120];
+```
+
+In the above function, we: 
+- Pass in a callback function, which is `(total, currentItem) => total * currentItem`.
+- Initialize total to `1` in the second argument.
+
+So what `.reduce()` will do, is it will once again go through every element in `arr` and apply the `callback` function to it. It then changes `total`, without actually changing the array itself. After it's done, it returns `total`.
+
+#### Summary 
+
+You've learnt about the three powerful array methods which are `map`, `filter` and `reduce`. They allow us to write shorter code that is more readable and less prone to bugs. 
+
+For a quick recap of these array methods, consider this picture which should visually explain them in terms of sandwiches:
+
+![Alt text](https://static.observableusercontent.com/thumbnail/bea194824f0d5842addcb7910bb488795c6f80f143ab5332b28a317ebcecd603.jpg)
```

Comment:
How about: 'example of filter, map and reduce methods being used to visually represent making a sandwich'. 

Let me know what you think about this. I think the image is worth including because it is a nice visual representation of what we've been talking about in the lesson and people will find it helpful.

Reply:
That sounds great to me!